### ISSUE TITLE
How to run this?

### ISSUE THREAD

**Issue by Mrinal18** [wangyirui/AD_Prediction#1]
hey, i was looking at your project and found it really interesting. Although there are no proper documentation in how to run it. can you kindly tell me that?

**Comment by wangyirui** [comment]
Glad to hear you are interested in our project. And we are sorry for the lack of description of running the code. Firstly, you should download the dataset (in our final report we have stated which dataset we use, so you can check the "data" part), and then running the code by: `python main_alextnet.py --optimizer Adam --learning_rate 1e-4 --save AlexNet-fine-tune-fc-last-conv-lr1e-4 --batch_size 16 --epochs 200 --gpuid 0`  It assumes you have one GPU (since running on CPU makes people crazy....). Feel free to tell us if you have further questions.

**Comment by Mrinal18** [comment]
Hey, thank you for such a fast response. 
Firstly, about data, i have my own data which is in raw format(example : Ad2_bet.nii.gz). For two classes(AD and NCI), so will it work for your code too?

**Comment by wangyirui** [comment]
Yes. In fact, we also do a two class classification.


On Apr 9, 2018, at 9:32 AM, Mrinal18 <notifications@github.com<mailto:notifications@github.com>> wrote:


Hey, thank you for such a fast response.
Firstly, about data, i have my own data which is in raw format(example : Ad2_bet.nii.gz). For two classes(AD and NCI), so will it work for your code too?

**Comment by Mrinal18** [comment]
That is really helpful, if possible can you send me a documentation as how to do it and run your program? that would be really helpful. and great work on the project. :)

**Comment by Mrinal18** [comment]
Hey, how do i classify this with 3d CNN in your repo?

**Comment by Mrinal18** [comment]
Also, i have one more confusion, how did you generate the test.txt, train.txt , test_2C_new.txt, etc.

basically how did you generate all the text files.

**Comment by wangyirui** [comment]
Sorry for the late reply. We are so busy these days.  For the first question, to run the code, you need to generate your own train.txt and test.txt (based on your own data samples and labels). Then prepare your data (you may need to let the program know where the data is, either by command line args or directly "hard-encode" in the program). Finally, you can run the code by the commands that I posted in the previous replies. For the second questions, we only implement 3D MRI classification by using autoencoder. The reason is that train a such a large network (e.g. AlextNet) by only using hundreds of samples is totally impossible! So we must construct a much simpler network (e.g. the autoencoder). To run 3D CNN you just need to run autoencoder.py (see the command I replied you earlier). For the third one, I guess we generate from an excel file (provided along with the nii data). 

BTW, if you think our project is useful and interesting ,would you mind giving us a star? :) Thanks!
